rich/syntax.py

```diff
@@ -484,8 +520,28 @@ def tokens_to_spans() -> Iterable[Tuple[str, Optional[Style]]]:
                 )
             if self.background_color is not None:
                 text.stylize(f"on {self.background_color}")
+
+        if self._stylized_ranges:
+            self._apply_stylized_ranges(code=code, text=text)
+
         return text
 
+    def stylize_range(
+        self, style: Style, start: SyntaxPosition, end: SyntaxPosition
```

Comment:
`style` should be a `StyleType` which is a union of a str and `Style`. The idea is that I can pass either `"bold red"` *or* a Style object.

You can convert a `StyleType` in to a `Style` by calling `console.get_style(style_type)`